src/core/Akka.TestKit/Internal/BlockingQueue.cs

```diff
@@ -22,7 +22,7 @@ namespace Akka.TestKit.Internal
     /// <typeparam name="T">The type of item to store.</typeparam>
     public class BlockingQueue<T>
     {
-        private readonly BlockingCollection<Positioned> _collection = new BlockingCollection<Positioned>();
+        private readonly BlockingCollection<Positioned> _collection = new BlockingCollection<Positioned>(new QueueWithAddFirst());
```

Comment:
`QueueWithAddFirst` is the backing `IProducerConsumerCollection` for the internal `BlockingCollection` that allows proper item insertion in the head of the queue.